docs/guide/module-configuration-patterns.md

```diff
@@ -0,0 +1,332 @@
+# Module Extension and Improvement
+
+One core feature of ApostropheCMS is its module system that allows you to customize and enhance functionality without modifying core code. This approach keeps your project maintainable and upgrade-friendly while giving you the flexibility to tailor the CMS to your needs. You will see this information partially repeated in several areas of our documentation including in the [reference section](/reference/module-api/module-overview.html#extend) because we feel it is so important for developers to understand.
+
+## Understanding Module Customization in ApostropheCMS
+
+When working with ApostropheCMS, it's important to understand that customizations should be made through files in your project's `/modules` folder rather than by directly modifying files in the `node_modules` directory.
+
+Modifying files in `node_modules` can create significant problems:
+- Changes are lost when you run `npm install` or `npm update`
+- Customizations won't carry over when deploying to different environments
+- Your changes won't be tracked in version control
+
+Instead, ApostropheCMS is designed to let you customize modules by creating corresponding files in your project's `/modules` folder. This way, your changes are:
+- Preserved during updates
+- Properly tracked in version control
+- Correctly deployed across environments
+- Applied consistently throughout your application
+
+## The Three Types of Module Customization
+
+ApostropheCMS provides three distinct ways to customize module behavior, each with its own purpose:
+
+### 1. Module Configuration
+
+**Use for:** Adjusting existing behavior using options that have been specifically built into modules.
```

Comment:
Options are often best set by following the modules/SAME-NAME/index.js pattern, and that is the most maintainable way to set them. So I think the "same filename" concept needs to be covered right from the beginning, while acknowledging that you can also set them in `app.js`.

Reply:
Rearranged and further explanation added.